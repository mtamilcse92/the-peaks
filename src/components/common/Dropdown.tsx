import React from "react";
import CX from "classnames";
import styles from "./Dropdown.module.scss"
import { ReactComponent as DownImage }  from "../../assets/arrow_drop_down_24px.svg";
import { ReactComponent as UpImage }  from "../../assets/arrow_drop_down_up_24px.svg";
const { useState } = React;

export type Props = {
    items?: {
        id: number | string
        label: string
    } [];
    onClick?: () => void
  };
  
const data = [{id: 0, label: "Istanbul, TR (AHL)"}, {id: 1, label: "Paris, FR (CDG)"}];

const Dropdown: React.FC<Props> = ({ items =data }) => {
  const [isOpen, setOpen] = useState(false);
  const [selectedItem, setSelectedItem] = useState<string | number | null>(null);
  
  const toggleDropdown = () => setOpen(!isOpen);
  
  const handleItemClick = (e: any): void => {
    setSelectedItem(Number(e.target.id))
    setOpen(false)
  }
  
  return (
    <div className={CX(styles.dropdown, {[styles.open]: isOpen })}>
      <div className={styles.dropdownHeader} onClick={toggleDropdown}>
        {items.find(item => item.id === selectedItem)?.label || "Select your destination"}
        {isOpen ? <UpImage/> : <DownImage />}
      </div>
      <div className={CX(styles.dropdownBody, {[styles.open]: isOpen })}>
        {items.map(item => (
          <div className={CX(styles.dropdownItem, {[styles.selected]: item.id === selectedItem })} onClick={handleItemClick} id={`${item.id}`}>
            {item.label}
          </div>
        ))}
      </div>
    </div>
  )
}

export default Dropdown;