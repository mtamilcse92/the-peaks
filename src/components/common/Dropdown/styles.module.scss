@import '../../../styles/index.scss';
@import '../../../styles/mixin.scss';
@import '../../../styles/variables.scss';

  .dropdown {
    width: 300px;
    margin-left: 8px;
    position: relative;
    background-color: white;
    border-bottom: 1px solid black;
    @include media(mobile) {
      flex-direction: column;
      margin: 8px;
      margin-left: 0px;
  }
}
.dropdown.open {
      border: 1px solid black;
  }
  
  .dropdownHeader {
    padding: 15px;
    color: black;
    cursor: pointer;
    display: flex;
    justify-content: space-between;
    align-items: center;
  }
  
  .dropdownBody {
    display: none;
  }
  
  .dropdownBody.open {
    display: block;
    left: -1px;
    position: absolute;
    z-index: 999;
    width: 100%;
    border: 1px solid black;
    border-top: none;
    background-color: white;
  }
  
  .dropdownItem {
      text-align: left;
      color: black;
    padding: 10px;
  }
  .dropdownItem.selected {
     background-color: #eaeaea;
  }
  
  .dropdownItem:hover {
    cursor: pointer;
  }
  
  .dropdownItemDot {
    opacity: 0;
    color: #91A5BE;
    transition: all .2s ease-in-out;
  }
  
  .dropdownItemDot.selected {
    opacity: 1;
  }
