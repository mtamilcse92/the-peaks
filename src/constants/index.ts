export const contentSort = [
    {
        id: 1,
        value: 'newest',
        label: 'Newest first'
    },
    {
        id: 2,
        value: 'oldest',
        label: 'Oldest first'
    },
    {
        id: 3,
        value: 'relevance',
        label: 'Most popular'
    }
]