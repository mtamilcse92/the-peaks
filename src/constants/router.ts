interface IPath { root: string, contents: string, contentDetail: string, bookmarks: string }

export const paths: IPath = {
    root: '/',
    contents: '/contents',
    contentDetail: '/contents/detail',
    bookmarks: '/bookmarks',
}