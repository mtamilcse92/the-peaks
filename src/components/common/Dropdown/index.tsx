import React from "react";
import CX from "classnames";
import { ReactComponent as DownImage }  from "../../../assets/arrow_drop_down_24px.svg";
import { ReactComponent as UpImage }  from "../../../assets/arrow_drop_down_up_24px.svg";
import { contentSort } from '../../../constants/index'
import styles from "./styles.module.scss"
const { useState } = React;

interface IItem {
  id: number | string
  label: string
  value: string
}

export type Props = {
    items?: IItem [];
    selected?: IItem
    onSelect: (item: IItem) => void
  };
  
const Dropdown: React.FC<Props> = ({ onSelect, selected, items = contentSort }) => {
  const [isOpen, setOpen] = useState(false);
  const [selectedItem, setSelectedItem] = useState<string | number | null>(selected?.id || null);
  
  const toggleDropdown = () => setOpen(!isOpen);
  
  const handleItemClick = (e: any): void => {
    setSelectedItem(Number(e.target.id))
    setOpen(false)
    const selectedValue = items.find(({ id }) => id === Number(e.target.id))
    
    if (selectedValue) {
      onSelect(selectedValue)
    }
  }
  
  return (
    <div className={CX(styles.dropdown, {[styles.open]: isOpen })}>
      <div className={styles.dropdownHeader} onClick={toggleDropdown}>
        {items.find(item => item.id === selectedItem)?.label || "Select your destination"}
        {isOpen ? <UpImage/> : <DownImage />}
      </div>
      <div className={CX(styles.dropdownBody, {[styles.open]: isOpen })}>
        {items.map(item => (
          <div key={item.id} className={CX(styles.dropdownItem, {[styles.selected]: item.id === selectedItem })} data-item={item} onClick={handleItemClick} id={`${item.id}`}>
            {item.label}
          </div>
        ))}
      </div>
    </div>
  )
}

export default Dropdown;