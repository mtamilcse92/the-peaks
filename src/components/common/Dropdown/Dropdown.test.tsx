import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import Dropdown from "./index";
import { contentSort } from '../../../constants/index'

describe("<Dropdown /> spec", () => {
  it("renders the Dropdown", () => {
    const Component = render(<Dropdown onSelect={jest.fn()} />);
    expect(Component).toMatchSnapshot();
  });

  it("should not render selected value", () => {
    const onClick = jest.fn();
    render(
      <Dropdown
        selected={contentSort[1]}
        onSelect={onClick}
      />
    );
    expect(screen.getAllByText(contentSort[1].label)).toHaveLength(2);
  });

  it("should fire the onselect props", () => {
    const onClick = jest.fn();
    const Component = render(
      <Dropdown
        onSelect={onClick}
      />
    );
    fireEvent.click(Component.getByText(contentSort[1].label));
    expect(onClick).toHaveBeenCalled();
  });
});
