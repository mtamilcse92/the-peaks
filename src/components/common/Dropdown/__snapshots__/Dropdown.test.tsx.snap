// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown /> spec renders the Dropdown 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dropdown"
      >
        <div
          class="dropdownHeader"
        >
          Select your destination
          <svg>
            arrow_drop_down_24px.svg
          </svg>
        </div>
        <div
          class="dropdownBody"
        >
          <div
            class="dropdownItem"
            data-item="[object Object]"
            id="1"
          >
            Newest first
          </div>
          <div
            class="dropdownItem"
            data-item="[object Object]"
            id="2"
          >
            Oldest first
          </div>
          <div
            class="dropdownItem"
            data-item="[object Object]"
            id="3"
          >
            Most popular
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dropdown"
    >
      <div
        class="dropdownHeader"
      >
        Select your destination
        <svg>
          arrow_drop_down_24px.svg
        </svg>
      </div>
      <div
        class="dropdownBody"
      >
        <div
          class="dropdownItem"
          data-item="[object Object]"
          id="1"
        >
          Newest first
        </div>
        <div
          class="dropdownItem"
          data-item="[object Object]"
          id="2"
        >
          Oldest first
        </div>
        <div
          class="dropdownItem"
          data-item="[object Object]"
          id="3"
        >
          Most popular
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
